apps/api/src/routers/guest.py

```diff
@@ -72,6 +76,9 @@ async def verify_guest(
     log.info("%s authenticated as guest.", email)
     guest = guest_auth.acquire_guest_identity(email)
 
-    res = RedirectResponse("/portal", status_code=status.HTTP_303_SEE_OTHER)
+    query = urlencode({"return_to": return_to})
+    res = RedirectResponse(
+        URL(path="/portal", query=query), status_code=status.HTTP_303_SEE_OTHER
+    )
```

Comment:
Suggestion: Since this is the step when authentication completes, it would be simpler to just redirect to the `return_to` value (and verify it starts with `/`) rather than redirecting again from `Portal.tsx` (similar to how in SAML ACS, the result is `RedirectResponse(relay_state, 303)`).